### PR TITLE
Update docs builder to latest version of Sphinx.

### PR DESCRIPTION
**Pull request by tswast**
SmartyPantsHTMLTranslator was removed. Also, writers were moved to
builders.

I also added Sphinx and the RTD theme to docs/requirements.txt (I did
also see them in `tox.ini`, but I'm used to installing the
requirements.txt files for developments.)

### CODE REVIEW

File: docs/custom_html_writer.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 """
 
 
-class CustomHTMLWriter(html.SmartyPantsHTMLTranslator):
+class CustomHTMLWriter(html.HTML5Translator):
```
Comment (dhermes) [inline docs/custom_html_writer.py]: @tswast I had to fix https://github.com/dhermes/bezier/blob/master/docs/custom_html_writer.py when Smarty Pants was removed. Any reason to pick `HTML5Translator` over `HTMLTranslator`?

**Review by dhermes — APPROVED** [review]
This LGTM, the AppVeyor build issues should be sorted out, but not in this PR. (It's just staleness, I know quite a bit more about Windows / AppVeyor than I did when I last worked on this.)

**Comment by dhermes** [comment]
@tswast Ping me if / when you want me to merge?

File: docs/custom_html_writer.py (hunk 1 above)
Comment (tswast) [reply docs/custom_html_writer.py]: Nope. Just the first one I saw. I can change to HTMLTranslator.

**Comment by dhermes** [comment]
@tswast Mind sending a PR with what results after just running `tox -e docs`?

Also the latest versions of Pylint seems to have broken `tox -e lint` with some of the latest modifications. I think I have tracked those updates in https://github.com/dhermes/bezier as well.

**Pull request merged**